2022/src/day16.scala

```diff
@@ -98,36 +98,39 @@ def bestPath(map: RoomsInfo, start: Id, valves: Set[Id], timeAllowed: Int): Int
   // we limit our options by only considering the still-closed valves
   // and `valves` has already culled any room with a flow value of 0 -- no point in considering these rooms!
 
-  val valvesArray = valves.toArray
-  val valveCount = valvesArray.length
-  val valvesIndicesLeft = Array.fill[Boolean](valveCount)(true)
-  val roomsByIndices = Array.tabulate(valveCount)(i => map.rooms(valvesArray(i)))
-
-  def recurse(current: Id, timeLeft: Int, totalValue: Int): Int =
+  val valvesLookup = IArray.from(valves)
+  val valveCount = valvesLookup.size
+  val _activeValveIndices = Array.fill[Boolean](valveCount + 1)(true) // add an extra valve for the initial state
+  def valveIndexLeft(i: Int) = _activeValveIndices(i)
+  def withoutValve(i: Int)(f: => Int) =
+    try
+      _activeValveIndices(i) = false
+      f
+    finally _activeValveIndices(i) = true
```

Comment:
The `try..finally` is also going to cost something on JS. And it's useless since we don't catch exceptions here until the array gets thrown away.

Reply:
alrighty

Reply:
I have reverted the changes anyway #264